portfolio/src/test/java/com/google/sps/TaskQueueTest.java

```diff
@@ -66,7 +65,153 @@ public void checkCorrectSortingShortestTaskFirst() {
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task2 =
+        new Task(
+            "Task 2",
+            "Second task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task3 =
+        new Task(
+            "Task 3",
+            "Third task",
+            SchedulerTestUtil.DURATION_10_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task4 =
+        new Task(
+            "Task 4",
+            "Fourth task",
+            SchedulerTestUtil.DURATION_15_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task5 =
+        new Task(
+            "Task 5",
+            "Fifth task",
+            SchedulerTestUtil.DURATION_20_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+
+    List<Task> taskList = Arrays.asList(task3, task1, task2, task5, task4);
+
+    List<Task> expected = Arrays.asList(task1, task3, task4, task5, task2);
+    List<Task> actual = new ArrayList<Task>();
+
+    TaskQueue taskQueue = new TaskQueue(taskList, HIGHEST_PRIORITY_FIRST);
+
+    int i = 0;
+    while (i++ < expected.size()) {
+      actual.add(taskQueue.peek());
+      taskQueue.remove();
+    }
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  /**
+   * Ensure that the PriorityQueue is correctly sorting the tasks from shortest to longest and
+   * highest priority to lowest using the HIGHEST_PRIORITY_FIRST comparator.
+   */
+  @Test
+  public void checkCorrectSortingHighestPriorityFirst() {
+    Task task1 =
+        new Task(
+            "Task 1",
+            "First task",
+            SchedulerTestUtil.DURATION_20_MINUTES,
+            SchedulerTestUtil.PRIORITY_TWO);
+    Task task2 =
+        new Task(
+            "Task 2",
+            "Second task",
+            SchedulerTestUtil.DURATION_20_MINUTES,
+            SchedulerTestUtil.PRIORITY_THREE);
+    Task task3 =
+        new Task(
+            "Task 3",
+            "Third task",
+            SchedulerTestUtil.DURATION_20_MINUTES,
+            SchedulerTestUtil.PRIORITY_FOUR);
+    Task task4 =
+        new Task(
+            "Task 4",
+            "Fourth task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_THREE);
+    Task task5 =
+        new Task(
+            "Task 5",
+            "Fifth task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_FOUR);
+    Task task6 =
+        new Task(
+            "Task 6",
+            "Sixth task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_FIVE);
+
+    List<Task> taskList = Arrays.asList(task3, task1, task2, task5, task4, task6);
+
+    List<Task> expected = Arrays.asList(task6, task3, task5, task2, task4, task1);
+    List<Task> actual = new ArrayList<Task>();
+
+    TaskQueue taskQueue = new TaskQueue(taskList, HIGHEST_PRIORITY_FIRST);
+
+    int i = 0;
+    while (i++ < expected.size()) {
```

Comment:
We can introduce a method to taskQueue like isEmpty(), and then here do,

while (!taskQueue.isEmpty()) {

Reply:
Done.